test/e2e/fixture-server.js

```diff
@@ -104,8 +104,11 @@ class FixtureServer {
   }
 
   loadJsonState(rawState, contractRegistry) {
-    const state = performStateSubstitutions(rawState, contractRegistry);
-    this._stateMap.set(CURRENT_STATE_KEY, state);
+    return new Promise((resolve) => {
```

Comment:
Are these changes needed? It looks like nothing async is happening here. Wrapping in a Promise would have no impact (except for making more confusing stack traces, if we get an error thrown in here)

Reply:
Also we should be avoiding Promise constructors _in general_, favoring async/await syntax. We only need Promise constructors for wrapping callback APIs, which isn't happening here.

But either way, I would expect this to not have any functional impact.

Reply:
whoa Mark, thank you! That was indeed not needed

https://github.com/user-attachments/assets/c447ec7d-2a63-4519-ad4c-26a437328a31

Reply:
Noted :pencil2: , thank you for your great insights as always, clarifying and full of wisdom :pray: